File: src/main/java/net/hectus/neobb/turn/person_game/block/PTBirchLog.java
package net.hectus.neobb.turn.person_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.turn.person_game.categorization.AttackCategory;
import org.bukkit.block.Block;

import java.util.List;

public class PTBirchLog extends BlockTurn implements AttackCategory {
    public PTBirchLog(NeoPlayer player) { super(player); }
    public PTBirchLog(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public void apply() {
        super.apply();
        player.nextPlayer().piercingDamage(2);
    }

    @Override
    public List<Class<? extends Turn<?>>> counteredBy() {
        return List.of(PTPinkCarpet.class, PTGreenCarpet.class);
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/block/PTBirchLog.kt
package net.hectus.neobb.modes.turn.person_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.Turn
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.modes.turn.person_game.categorization.AttackCategory
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block
import kotlin.reflect.KClass

class PTBirchLog(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), AttackCategory {
    override fun apply() {
        player!!.nextPlayer().damage(2.0, true)
    }

    override fun counteredBy(): List<KClass<out Turn<*>>> {
        return listOf(PTPinkCarpet::class, PTGreenCarpet::class)
    }
}
